### PR TITLE
#311 Clean project structure

### PR DESCRIPTION
**Pull request by AndrewJack**
**Fixes**: #311 

Organised the project structure of this library.

Moving of the ios directory is a breaking change so will need to be noted in the release notes.

Not sure on where the `.podspec` should go?

Separated into small commits for easy review - let me know if you'd like it squashed

**Other changes**:
- Updated gitignore & removed from example
- aligned example with project's react native version `react-native upgrade`
- add .npmignore - exclude example directory & remove "files" from `package.json`

After this PR I have a few other contributions for this project that've made on our fork, such as upgrading to latest react-native, and using exoplayer for the android player (Which should resolve most of the issues on this library).

You can view the WIP here https://github.com/drivetribe/react-native-video/tree/exoplayer

cc @mattapperson

### CODE REVIEW

**Comment by mattapperson** [comment]
@AndrewJack sorry I was out sick so I missed this. Could you please rebase and then I will merge?

File: android/src/main/java/com/brentvatne/react/ReactVideoView.java [hunk 1]
```diff
@@ -258,7 +258,11 @@ public void setSrc(final String uriString, final String type, final boolean isNe
         mEventEmitter.receiveEvent(getId(), Events.EVENT_LOAD_START.toString(), event);
 
         // not async to prevent random crashes on Android playback from local resource due to race conditions
-        prepare(this);
+        try {
+          prepare(this);
+        } catch (Exception e) {
+          e.printStackTrace();
+        }
```
Comment (AndrewJack) [inline android/src/main/java/com/brentvatne/react/ReactVideoView.java:265]: Got a compile error without. didn't happen before has something changed in master?

**Comment by AndrewJack** [comment]
@mattapperson rebased

File: android/src/main/java/com/brentvatne/react/ReactVideoView.java (hunk 1 above)
Comment (mattapperson) [reply android/src/main/java/com/brentvatne/react/ReactVideoView.java:265]: Yea, this is being worked on

**Comment by corbt** [comment]
@AndrewJack would love to see that PR for ExoPlayer, the poor performance and occasional crashes with the current Android player are causing issues!

**Comment by AndrewJack** [comment]
@corbt sure it's on my list, I'll hopefully get round to it soon.

**Pull request merged**